pgsql/pc_editor.c

```diff
@@ -10,85 +10,117 @@
 #include "pc_pgsql.h"	   /* Common PgSQL support for our type */
 
 Datum pcpatch_setpcid(PG_FUNCTION_ARGS);
+Datum pcpatch_transform(PG_FUNCTION_ARGS);
 
 
-static bool
-pcpatch_schema_same_dimensions(const PCSCHEMA *s1, const PCSCHEMA *s2)
+static SERIALIZED_PATCH *
+pcpatch_set_schema(SERIALIZED_PATCH *serpa,
+				  PCSCHEMA *oschema, PCSCHEMA *nschema, float8 def)
 {
-	size_t i;
-
-	if ( s1->ndims != s2->ndims )
-		return false;
-
-	for ( i = 0; i < s1->ndims; i++ )
-	{
-		PCDIMENSION *s1dim = s1->dims[i];
-		PCDIMENSION *s2dim = s2->dims[i];
-
-		if ( strcasecmp(s1dim->name, s2dim->name) != 0 )
-			return false;
-
-		if ( s1dim->interpretation != s2dim->interpretation )
-			return false;
-	}
-
-	return true;
-}
-
-
-PG_FUNCTION_INFO_V1(pcpatch_setpcid);
-Datum pcpatch_setpcid(PG_FUNCTION_ARGS)
-{
-	PCPATCH *paout = NULL;
 	SERIALIZED_PATCH *serpatch;
-	SERIALIZED_PATCH *serpa = PG_GETARG_SERPATCH_P(0);
-	int32 pcid = PG_GETARG_INT32(1);
-	float8 def = PG_GETARG_FLOAT8(2);
-	PCSCHEMA *old_schema = pc_schema_from_pcid(serpa->pcid, fcinfo);
-	PCSCHEMA *new_schema = pc_schema_from_pcid(pcid, fcinfo);
+	PCPATCH *paout;
 
-	if ( pcpatch_schema_same_dimensions(old_schema, new_schema) )
+	if ( pc_schema_same_dimensions(oschema, nschema) )
 	{
-		// old_schema and new_schema have the same dimensions at the same
-		// positions, so we can take a fast path avoid the point-by-point,
-		// dimension-by-dimension copying
+		// oschema and nschema have the same dimensions at the same
+		// positions, so we can take a fast path and avoid the
+		// point-by-point dimension-by-dimension copying
 
-		if ( old_schema->compression == new_schema->compression )
+		if ( oschema->compression == nschema->compression )
 		{
 			// no need to deserialize the patch
 			serpatch = palloc(serpa->size);
 			if ( ! serpatch )
-				PG_RETURN_NULL();
+				return NULL;
 			memcpy(serpatch, serpa, serpa->size);
-			serpatch->pcid = pcid;
-			PG_RETURN_POINTER(serpatch);
+			serpatch->pcid = serpa->pcid;
```

Comment:
Good catch. Yes, that's covered: https://github.com/LI3DS/pointcloud/blob/36a2fc52284be39858263cb4e2ef4330ca3629d0/pgsql/sql/pointcloud.sql#L417-L422.